src/plugins/embeddable/public/lib/embeddables/compatibility/legacy_embeddable_to_api.ts

```diff
@@ -202,7 +209,13 @@ export const legacyEmbeddableToApi = (
 
       subscriptions.add(
         embeddable.getInput$().subscribe(() => {
-          if (!compareFilters(embeddable.filters$.getValue() ?? [], embeddable.getFilters())) {
+          if (
```

Comment:
default `comparatorOptions` is, which means that compareOptions will return true (no changes) when there are any changes in those keys. Which is not what we want. 
```
comparators = defaults(comparatorOptions || {}, {
    index: false,
    state: false,
    negate: false,
    disabled: false,
    alias: false,
  });
```